policies/recipes/steps.go

```diff
@@ -80,18 +83,26 @@ func stepCopyFile(step *agentendpointpb.SoftwareRecipe_Step_CopyFile, artifacts
 		return err
 	}
 	defer reader.Close()
-	writer, err := os.OpenFile(dest, os.O_TRUNC|os.O_WRONLY|os.O_CREATE, permissions)
+
+	tmp, err := ioutil.TempFile(filepath.Base(dest), ".tmp*")
 	if err != nil {
 		return err
 	}
-	defer writer.Close()
 
-	_, err = io.Copy(writer, reader)
-	if err != nil {
+	if _, err = io.Copy(tmp, reader); err != nil {
+		tmp.Close()
 		return err
 	}
 
-	return nil
+	if err := tmp.Chmod(permissions); err != nil {
+		return err
+	}
+
+	if err := tmp.Close(); err != nil {
+		return err
+	}
```

Comment:
Would an AtomicFileCopy be helpful? The only difference is io.Copy vs. io.Write.

Reply:
Not needed by any other function right now, will add if I find other uses